plugins/catalog-backend/src/next/DefaultLocationService.ts

```diff
@@ -41,30 +42,45 @@ export class DefaultLocationService implements LocationService {
         apiVersion: 'backstage.io/v1alpha1',
         kind: 'Location',
         metadata: {
-          name: `${spec.type}:${spec.target}`,
+          name: locationSpecToMetadataName({
+            type: spec.type,
+            target: spec.target,
+          }),
           namespace: 'default',
           annotations: {
             [LOCATION_ANNOTATION]: `${spec.type}:${spec.target}`,
             [ORIGIN_LOCATION_ANNOTATION]: `${spec.type}:${spec.target}`,
           },
         },
         spec: {
-          location: { type: spec.type, target: spec.target },
+          type: spec.type,
+          target: spec.target,
         },
       };
-      const processed = await this.orchestrator.process({
-        entity,
-        eager: true,
-        state: new Map(),
-      });
-      if (processed.ok) {
-        return {
-          location: { ...spec, id: `${spec.type}:${spec.target}` },
-          entities: [processed.completedEntity],
-        };
+      const unprocessedEntities: Entity[] = [entity];
+      const entities: Entity[] = [];
+      const state = new Map(); // ignored
+      while (unprocessedEntities.length) {
+        const currentEntity = unprocessedEntities.pop();
+        if (!currentEntity) {
+          continue;
+        }
+        const processed = await this.orchestrator.process({
+          entity: currentEntity,
+          state,
+        });
+        if (processed.ok) {
+          unprocessedEntities.push(...processed.deferredEntities);
+          entities.push(processed.completedEntity);
+        } else {
+          throw Error(processed.errors.join(', '));
```

Comment:
yeah it's not ideal. The catalog import definitely need some extra love. Currently fails because it expects entities to be returned from the non dryRun createLocation. There's a todo for that added to the catalog import plugin.